tools/doxygen/doxygen.sh

```diff
@@ -0,0 +1,127 @@
+#!/bin/bash
+
+LAMMPSBASE=${PWD} ;
+LAMMPSSRC="${LAMMPSBASE}/src" ;
+LAMMPSDOC="${LAMMPSBASE}/doc" ;
+LAMMPSDOCHTML="${LAMMPSDOC}/html" ;
+LAMMPSDEVELOPERDOC="${LAMMPSBASE}/doc/src/Developer" ;
+LAMMPSDOCTMP="/tmp/lammps-docs-*" ;
+LAMMPSDOCMAKE="${LAMMPSDOC}/Makefile" ;
+LAMMPSDOXYGEN="${LAMMPSBASE}/tools/doxygen" ;
+LAMMPSDOXYGENDOC="${LAMMPSDOXYGEN}/doc" ;
+LAMMPSDOXYGENDOCMANUAL="${LAMMPSDOXYGEN}/doc/html/Manual" ;
+
+LAMMPSDOXYFILE=${LAMMPSDOXYGEN}/Doxyfile.lammps ;
+DOXYFILE=${LAMMPSDOXYGEN}/Doxyfile ;
+
+LAMMPSDEVELOPERDOXFILE=${LAMMPSDOXYGEN}/Developer.dox.lammps ;
+DEVELOPERDOXFILE=${LAMMPSDOXYGEN}/Developer.dox ;
+
+FIG2DEV=`which fig2dev` ;
+DOXYGEN=`which doxygen` ;
+
+
+error=0;
+
+
+if [[ -d ${LAMMPSSRC} && -d ${LAMMPSDEVELOPERDOC} && -d ${LAMMPSDOC} && -f ${LAMMPSDOCMAKE} && -d ${LAMMPSDOXYGEN} && -f ${LAMMPSDOXYFILE} ]] ;
+ then
+   cd ${LAMMPSSRC} ;
+   make no-all ;
+   cd ${LAMMPSBASE} ;
+ else
+  error=1 ;
+  echo "Cannot configure LAMMPS sources - Please run doxygen.sh from the LAMMPS root directory."
+ fi ;
+
+
+if [ ${error} == 0 ] ;
+  then
+    if [[ ! -x ${FIG2DEV} ]] ;
+      then
+        error=1 ;
+        echo "${FIG2DEV} not found - Please install ${FIG2DEV} for Your operating system." ;
+      fi ;
+  fi ;
+
+
+if [ ${error} == 0 ] ;
+  then
+    if [[ -d ${LAMMPSDEVELOPERDOC} ]] ;
+      then
+        ${FIG2DEV} -L png ${LAMMPSDEVELOPERDOC}/classes.fig > tools/doxygen/classes.png ;
+        ${FIG2DEV} -L eps ${LAMMPSDEVELOPERDOC}/classes.fig > tools/doxygen/classes.eps ;
+      else
+        error=1 ;
+        echo "LAMMPS developer documentation not found - Please control Your LAMMPS installation." ;
+      fi;
+  fi;
+
+
+if [ ${error} == 0 ] ;
+  then
+    if [[ ! -x ${DOXYGEN} ]] ;
+      then
+        error=1;
+        echo "doxygen not found - Please install doxygen for Your operating system." ;
+      fi ;
+  fi ;
+
+
+if [ ${error} == 0 ] ;
+  then
+    if [[ -d ${LAMMPSSRC} && -f ${LAMMPSDOXYFILE} && -f ${LAMMPSDEVELOPERDOXFILE} ]] ;
+      then
+        ICS=' ';
+        read -r -a array < ${LAMMPSSRC}/version.h ;
```

Comment:
That seems a bit complicated, something like `sed -e 's/.*"//' -e 's/".*//' -e 's/ //' version.h` should to it all at once.

Reply:
I am not too much a "sed" expert.